index.d.ts

```diff
@@ -0,0 +1,110 @@
+import { Action, ActionCreator, AnyAction, StoreEnhancer, Store } from 'redux';
+
+export interface StoreCreator {
+  <S, A extends Action>(
+    reducer: LoopReducer<S, A>,
+    preloadedState: S,
+    enhancer: StoreEnhancer<S>
+  ): Store<S>;
+}
+
+export type Loop<S, A extends Action> = [S, CmdType<A>];
+
+export interface LoopReducer<S, A extends Action> {
+  (state: S | undefined, action: A): S | Loop<S, A>;
+}
+
+declare interface LiftedLoopReducer<S, A extends Action> {
+  (state: S | undefined, action: A): Loop<S, A>;
+}
+
+export interface NoneCmd {
+  type: 'NONE';
+}
+
+export interface ActionCmd<A extends Action> {
+  type: 'ACTION';
+  actionToDispatch: A;
+}
+
+export interface BatchCmd<A extends Action> {
+  type: 'BATCH';
+  cmds: CmdType<A>[];
+}
+
+export interface MapCmd<A extends Action> {
+  type: 'MAP';
+  tagger: ActionCreator<A>;
+  nestedCmd: CmdType<A>;
+  args: any[];
+}
+
+export interface RunCmd<A extends Action> {
+  type: 'RUN';
+  func: Function;
+  options: {
+    args: any[];
+    failActionCreator: ActionCreator<A>;
+    successActionCreator: ActionCreator<A>;
+  };
+}
+
+export interface SequenceCmd<A extends Action> {
+  type: 'Sequence';
+  cmds: CmdType<A>[];
+}
+
+export type CmdType<A extends Action> =
+  | ActionCmd<A>
+  | BatchCmd<A>
+  | MapCmd<A>
+  | NoneCmd
+  | RunCmd<A>
+  | SequenceCmd<A>;
+
+declare function install<S>(): StoreEnhancer<S>;
+
+declare function loop<S, A extends Action>(
+  state: S,
+  loop: CmdType<A>
+): Loop<S, A>;
+
+declare class Cmd {
+  static readonly none: NoneCmd;
+  static action: <A extends Action>(action: A) => ActionCmd<A>;
+  static batch: <A extends Action>(cmds: CmdType<A>[]) => BatchCmd<A>;
+  static readonly dispatch: (action: Action) => void;
+  static readonly getState: any;
+  static map: <A extends Action, B extends Action>(
+    cmd: CmdType<B>,
+    tagger: (subAction: B) => A,
+    args?: any[]
+  ) => MapCmd<A>;
+  static run: <A extends Action>(
+    f: Function,
+    options?: {
+      args?: any[];
+      failureActionCreator?: ActionCreator<A>;
+      successActionCreator?: ActionCreator<A>;
+    }
+  ) => RunCmd<A>;
+  static sequence: <A extends Action>(cmds: CmdType<A>[]) => SequenceCmd<A>;
+}
+
+export interface StrictReducerMapObject<T, A extends Action> {
```

Comment:
could this just be ReducerMapObject? Strict makes it sound like there's a non-strict version

Reply:
👍